scripts/fuzz_opt.py

```diff
@@ -1164,23 +1200,30 @@ def handle_pair(self, input, before_wasm, after_wasm, opts):
             # happens, which is something like "[fuzz-exec] calling bar", and
             # it is unique since it contains the function being called.
             call_line = before[call_start:call_end]
-            # remove everything from that call line onward.
-            lines_pre = before.count(os.linesep)
-            before = before[:call_start]
-            lines_post = before.count(os.linesep)
-            print(f'ignoring code due to trap (from "{call_line}"), lines to compare goes {lines_pre} => {lines_post} ')
-
-            # also remove the relevant lines from after.
-            if call_line not in after:
-                # the normal run hit a trap, and the tnh run hit a host
-                # limitation that forces us to ignore this run. for example,
-                # after running tnh we may end up doing an unbounded number of
-                # allocations, if that is what the program normally does (and
-                # the normal run only avoided that by trapping).
-                assert IGNORE in after
-                return
-            after_index = after.index(call_line)
-            after = after[:after_index]
+            trapping_export = get_export_from_call_line(call_line)
+
+            # now that we know the trapping export, we can leave only the safe
+            # ones that are before it
```

Comment:
There's no good way to remove all the trapping exports but leave all the non-trapping exports in place?

Reply:
Unfortunately no (before or after this PR) because of global effects. If we call exports `a,b,c` and only `b` traps then it is possible that calling `a,c` will now trap in `c` when it didn't before. In other words, we have a single linear stream of execution of the exports in order, and once that traps we have to stop considering code from that point onward - running any of the later code is risky.

Reply:
Makes sense. I guess we could run in a loop, always just removing the first trapping function until there are no more traps, but that sounds expensive.